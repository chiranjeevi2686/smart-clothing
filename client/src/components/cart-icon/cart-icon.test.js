import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CartIcon from './cart-icon.component';

const defaultStore = {
    cart: {
        cartItems: [{
            id: 19,
            imageUrl: "https://i.ibb.co/mJS6vz0/blue-jean-jacket.png",
            name: "Blue Jean Jacket",
            price: 90,
            quantity: 1
        }],
        hidden: true,
        payment: null
    }
};
const mockStore = configureStore();

describe('CartIcon component', () => {

    it('should render cart icon', () => {

        const { asFragment } = render(<Provider store={mockStore(defaultStore)}>
            <CartIcon />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        const cartIconButton = screen.getByTestId('shopping-icon');
        fireEvent.click(cartIconButton);
    })
})
