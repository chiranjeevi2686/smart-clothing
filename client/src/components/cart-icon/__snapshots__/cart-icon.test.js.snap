// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should render cart icon 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ikjkaB"
  >
    <svg
      class="sc-gsTCUz ilxHWr"
      data-test-id="shopping-icon"
    >
      shopping-bag.svg
    </svg>
    <span
      class="sc-dlfnbm gkfeNm"
    >
      1
    </span>
  </div>
</DocumentFragment>
`;
